### PR TITLE
Add scalar angle multiplication

### PR DESCRIPTION
**Pull request by cquiroz**

### CODE REVIEW

File: modules/core/shared/src/main/scala/lucuma/core/math/Angle.scala [hunk 1]
```diff
@@ -300,6 +300,20 @@ object Angle extends AngleOptics {
...
+    * @group Operations
+    */
+    def *(a: Long): Angle =
+      Angle.fromMicroarcseconds(toMicroarcseconds * a)
+
+    /**
+    * Scalar multiplication of this angle and double precission number `a`. Approximate
+    * @group Operations
+    */
+    def *?(a: Double): Angle =
```
Comment (cquiroz) [inline modules/core/shared/src/main/scala/lucuma/core/math/Angle.scala:314]: I made it `*?` to denote that is approximate. not sure if it is a good idea, WDYT?

File: modules/core/shared/src/main/scala/lucuma/core/math/Angle.scala [hunk 2]
```diff
@@ -300,6 +300,20 @@ object Angle extends AngleOptics {
     def -(a: Angle): Angle =
       Angle.fromMicroarcseconds(toMicroarcseconds - a.toMicroarcseconds)
 
+    /**
+    * Scalar multiplication of this angle and and natural number `a`. Exact
+    * @group Operations
+    */
+    def *(a: Long): Angle =
+      Angle.fromMicroarcseconds(toMicroarcseconds * a)
```
Comment (tpolecat) [inline modules/core/shared/src/main/scala/lucuma/core/math/Angle.scala:308]: Unfortunately this overflows when `a` gets large. You may need to go to `BigDecimal` and renormalize it back into 0..360° before constructing the return value.

```
scala> Angle.Angle90 * 4000000
val res14: lucuma.core.math.Angle = 0
                                                                                                                                                                            
scala> Angle.Angle90 * 40000000
val res15: lucuma.core.math.Angle = 230290448384
```

File: modules/core/shared/src/main/scala/lucuma/core/math/Angle.scala (hunk 1 above)
Comment (tpolecat) [reply modules/core/shared/src/main/scala/lucuma/core/math/Angle.scala:314]: We just document it when it's inexact. Pretty much anything involving `Double` is inexact by definition.

File: modules/core/shared/src/main/scala/lucuma/core/math/Angle.scala (hunk 2 above)
Comment (cquiroz) [reply modules/core/shared/src/main/scala/lucuma/core/math/Angle.scala]: Thanks I pushed a new commit doing as you suggest and some tests

File: modules/core/shared/src/main/scala/lucuma/core/math/Angle.scala (hunk 1 above)
Comment (cquiroz) [reply modules/core/shared/src/main/scala/lucuma/core/math/Angle.scala]: I reverted to `*`

**Review by tpolecat — APPROVED** [review]

**Pull request merged**